ui/src/app/views/notebook-redirect/component.ts

```diff
@@ -210,6 +211,9 @@ export class NotebookRedirectComponent implements OnInit, OnDestroy {
   setNotebookNames(): void {
     this.notebookName =
       decodeURIComponent(this.route.snapshot.params['nbName']);
+    this.notebookName = this.notebookName.replace('#', '\\#');
```

Comment:
Did you figure out why these two characters in particular were causing a problem? If so, can you throw a comment here?

Reply:
I suspect its because it is a # fragment identifier. But i did not get a chance to read more. I'll confirm and add comments